Python 3 tests can't run because python-dev and python3.6-dev now conflict

```
Step 7/28 : RUN apt-get -y install libffi-dev python3.6 python3.6-dev python-dev python3-dev python-pip python3-pip libcurl4-openssl-dev libssl-dev wget curl openssh-server mesos=1.0.1-2.0.94.ubuntu1604 nodejs rsync screen && apt-get clean && rm -rf /var/lib/apt/lists/*
 ---> Running in 20f21279c094
Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 python3.6 : Breaks: python3-dev (< 3.6.5~rc1-1) but 3.5.1-3 is to be installed
E: Unable to correct problems, you have held broken packages.
The command '/bin/sh -c apt-get -y install libffi-dev python3.6 python3.6-dev python-dev python3-dev python-pip python3-pip libcurl4-openssl-dev libssl-dev wget curl openssh-server mesos=1.0.1-2.0.94.ubuntu1604 nodejs rsync screen && apt-get clean && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100
Makefile:202: recipe for target 'docker' failed
make: *** [docker] Error 100
```

We can't have both these packages installed at the same time apparently, at least with the versions currently in the repos.

See also maybe: #2408

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-357)
┆Issue Number: TOIL-357

--- a/docker/Dockerfile.py
+++ b/docker/Dockerfile.py
@@ -24,7 +24,6 @@ dependencies = ' '.join(['libffi-dev',  # For client side encryption for 'azure'
                          'python3.6',
                          'python3.6-dev',
                          'python-dev',  # For installing Python packages with native code
-                         'python3-dev',
                          'python-pip',  # Bootstrap pip, but needs upgrading, see below
                          'python3-pip',
                          'libcurl4-openssl-dev',

